std/assembly/internal/typedarray.ts

```diff
@@ -231,3 +232,47 @@ export function EVERY<TArray extends TypedArray<T>, T>(
   }
   return true;
 }
+
+@inline
+export function FOR_EACH<TArray extends TypedArray<T>, T>(
+  array: TArray,
+  callbackfn: (value: T, index: i32, array: TArray) => void,
+): void {
+  var buffer = array.buffer;
+  var length = array.length;
+  var byteOffset = array.byteOffset;
+
+  for (let i = 0; i < length; i++) {
+    callbackfn(LOAD<T>(buffer, i, byteOffset), i, array);
+  }
+}
+
+@inline
+export function FILTER<TArray extends TypedArray<T>, T>(
+  array: TArray,
+  callbackfn: (value: T, index: i32, array: TArray) => bool,
+): TArray {
+  var backingBuffer = new ArrayBuffer(sizeof<T>());
```

Comment:
Should probably use `allocateUnsafe` directly. Safes a few checks, and doesn't zero memory.

Reply:
Okay. For now it seems wise to call `allocateUnsafe()` directly.